src/PDFPasswordForm.tsx

```diff
@@ -0,0 +1,164 @@
+import React, {useState, useRef, useMemo, type ChangeEvent, type FormEventHandler} from 'react';
+import PropTypes from 'prop-types';
+
+import {pdfPasswordFormStyles as styles} from './styles';
+
+type Props = {
+    isPasswordInvalid: boolean;
+    isFocused: boolean;
+    onSubmit: (password: string) => void;
+    onPasswordUpdated: (password: string) => void;
+    onPasswordFieldFocused: (isFocused: boolean) => void;
+};
+
+const propTypes = {
+    /** If the submitted password is invalid (show an error message) */
+    isPasswordInvalid: PropTypes.bool,
+
+    /** Should focus to the password input  */
+    isFocused: PropTypes.bool.isRequired,
+
+    /** Notify parent that the password form has been submitted */
+    onSubmit: PropTypes.func,
+
+    /** Notify parent that the password has been updated/edited */
+    onPasswordUpdated: PropTypes.func,
+
+    /** Notify parent that a text field has been focused or blurred */
+    onPasswordFieldFocused: PropTypes.func,
+};
+
+const defaultProps = {
+    isPasswordInvalid: false,
+    onSubmit: () => {},
+    onPasswordUpdated: () => {},
+    onPasswordFieldFocused: () => {},
+};
+
+// @ts-expect-error TODO: Comment
+const isSafari = Boolean(window.safari);
```

Comment:
In E/App we seem to have a more complex logic to detect if the browser is Safari, did you check it? Not sure if this is enough.

Reply:
Yeah, I see. AFAIK, there is not another way than copy-paste it.

Reply:
Done - ea3c5c0.